src/compiler/file_uploader/upload_file.ts

```diff
@@ -69,3 +74,16 @@ function calculatePercentComplete(
     }
     return (bytesComplete / Math.max(fileSize, 1)) * 100;
 }
+
+async function shouldBeUploaded(
+    srcPath: string,
+    destPath: string,
+    actor: UploaderActor
+): Promise<boolean> {
+    const localHash = (await hashFile(srcPath)).toString('hex');
+    const canisterHash = await actor.get_file_hash(destPath);
+    if (canisterHash.length === 0) {
```

Comment:
Rename canisterHash to canisterHashOption

Reply:
done